### ISSUE TITLE
Tarball from PYPI is missing requirements.txt

### ISSUE THREAD

**Issue by yurivict** [pyTooling/pyTooling#82]
```
[pyTooling.Packaging] Could not import from 'pyTooling.*'!
[pyTooling.MetaClasses] Could not import from 'pyTooling.*'!
[pyTooling.MetaClasses] Could not import from 'pyTooling.*'!
[pyTooling.Licensing] Could not import from 'pyTooling.*'!
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "setup.py", line 47, in <module>
    DescribePythonPackageHostedOnGitHub(
  File "/usr/ports/devel/py-pyTooling/work-py39/pyTooling-2.12.3/pyTooling/Packaging/__init__.py", line 451, in DescribePythonPackageHostedOnGitHub
    DescribePythonPackage(
  File "/usr/ports/devel/py-pyTooling/work-py39/pyTooling-2.12.3/pyTooling/Packaging/__init__.py", line 333, in DescribePythonPackage
    requirements = list(set(loadRequirementsFile(requirementsFile, debug=True)))
  File "/usr/ports/devel/py-pyTooling/work-py39/pyTooling-2.12.3/pyTooling/Packaging/__init__.py", line 107, in loadRequirementsFile
    with requirementsFile.open("r") as file:
  File "/usr/local/lib/python3.9/pathlib.py", line 1252, in open
    return io.open(self, mode, buffering, encoding, errors, newline,
  File "/usr/local/lib/python3.9/pathlib.py", line 1120, in _opener
    return self._accessor.open(self, flags, mode)
FileNotFoundError: [Errno 2] No such file or directory: 'requirements.txt'
```

**Comment by Paebbels** [comment]
Is installing from wheel also an option under FreeBSD?

------------
Can / should my pipelines in GitHub Actions also test under FreeBSD?

**Comment by yurivict** [comment]
> Is installing from wheel also an option under FreeBSD?

Not in the port context.

**Comment by Paebbels** [comment]
I don't think the source distribution method supports dependency analysis and resolution.

The list of dependencies is handed over to the build backend (setuptools). This includes all necessary information in a wheel package, but apparently does not for source distributions.

Some people include `requirements.txt` as an additional file, but this is at least for wheel packages not correct. For source distributions, it looks like a workaround.

I don't see an option to discover in a `setup.py`, in what context is is called or what target format is selected.

-----------
Why are wheels not supported?  
What about eggs?

**Comment by yurivict** [comment]
> I don't think the source distribution method supports dependency analysis and resolution.

Dependencies are contained in setup.py or pyproject.toml.
Python tools like 'build' are able to find the list of dependencies and build the wheel or binaries purely from the source tarball.

**Comment by Paebbels** [comment]
Currently the dependencies are handed over to `setup()` in `setup.py`. It reads all of them from `requirements.txt` and if necessary even recursively.

The CI server uses `build` to package.

**Comment by Paebbels** [comment]
OK, I see the CI server doesn't use `build`, that's not good. I'll investigate ...

![image](https://user-images.githubusercontent.com/956109/221041823-a02e5e8c-b263-4adb-bb22-bcbda80624aa.png)

**Comment by yurivict** [comment]
The ```requirements.txt``` file isn't present in the tarball.

**Comment by Paebbels** [comment]
The latest binary distribution (wheel) and source distribution (sdist) are right now building using `build` (see release pipeline: https://github.com/pyTooling/pyTooling/actions/runs/4383455189) and will be pushed to PyPI in a few minutes.

I also investigated and checked the sdist package (`*.tar.gz`). The requirements are listed in the archive in `pyTooling-2.13.0\pyTooling.egg-info\requires.txt`. As you'll see, all information is contained in the sdist package.

I assume, you're using `setup.py` to install the package, but `setup.py` is meant to package code, not to install code. Misusing `setup.py` for a development installation, you need the full Git repository, which would also bring the `requirements.txt` files.

As to my research, FreeBSD has a `py-wheel` package supporting installation via PIP and wheel packages. This is the recommended and platform independent way of installing Python packages on Linux, MacOS, Windows.

Unfortunately, FreeBSD is not offered at GitHub Actions, so I can't test it.

To summarize:
* I don't think adding `requirements.txt` is the right way.
* I'm still interested in supporting you getting it to work on FreeBSD.
* Please recheck how to use wheels on FreeBSD or please explain why wheels are not working out in your situation.

**Comment by Paebbels** [comment]
@yurivict how can we solve this one for your needs?

**Comment by yurivict** [comment]
Please close it if it builds fine with ```build```.
